docs/userguide/icu4j/index.md

```diff
@@ -68,41 +67,122 @@ There are a few different ways to download the ICU4J releases.
+| `main/localespi`  | `icu4j-localespi`    | `core` <br/> `collate` | n/a | Implementation of various locale-sensitive service providers defined in   `java.text.spi`   and   `java.util.spi`  in J2SE 6.0 or later Java releases. This sub-component is shipped as `icu4j-localespi.jar`. |
+| `main/regiondata` | `regiondata`   | `core` | n/a | No public API classes. Provides access to region display data. This sub-component is packaged as a part of `icu4j.jar`. |
+| `main/translit`   | `translit`     | `core` | `com.ibm.icu.text` | Transliterator APIs and implementation. This sub-component is packaged as a part of `icu4j.jar`. |
+
+### ICU4J unit test files
+
+Unit test and integration tests are grouped together in the same Maven submodule as the runtime class code
+for the same component.
+The test source code lives in the standard Maven directory structure,
+`src/test/java` for test sources,
+and `src/test/resources` for test-specific non-source files.
+
+> :point_right: **Note**: Tests that depend on multiple sub-components must be placed
+in the `main/common_tests` directory and run as integration tests.
+
+The Maven build enforces isolation vis-a-vis dependencies for tests within a multi-module project such that
+unit tests must only test code within its same submodule.
+When a test depends on runtime code from multiple modules, it cannot exist in any of those modules,
+or else it would create a circular dependency in the Maven build.
+Instead, such tests should be refactored into a separate component
+and designated as integration tests.
+
+| Path                                      | Sub-component Name              | Runtime Dependencies                                                                                         | Description                                     |
+|-------------------------------------------|---------------------------------|--------------------------------------------------------------------------------------------------------------|-------------------------------------------------|
+| `main/common_tests`                        | `common_tests`             | `core` <br/> `currdata` <br/> `translit` <br/> `langdata` <br/> `collate` <br/> `regiondata` <br/>                                                                 | Sub-component for integration tests that depend on runtime code of multiple components. |
+
+
+### Others
+
+<table>
+    <tr>
+        <th>Path</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><code>main/shared</code></td>
+        <td>(Obsolete?) Files shared by ICU4J sub-components under the <code>main</code> directory including: <ul> <li>ICU4J runtime data archive (icudata.jar).</li> <li>ICU4J unit test data archive (testdata.jar).</li> <li>Shared Ant build script and configuration files.</li> <li>License files.</li></ul> </td>
```

Comment:
+10
Obsolete, can remove the whole line.

Reply:
Done.